### PR TITLE
Swimming in water

### PR DESCRIPTION
**Pull request by CrimsonShrike**
## Description of changes

Adds a barebones implementation of swimming. "Swimming" is the ability to overcome gravity as long as there is enough fluid to feasibly swim in.

Also makes falling onto deep fluid pools result in no damage.

Additionally changes breathing code to make swimming easier.

## Why and what will this PR improve

No longer do people walk at the bottom of pools, slowly drowning.

## Authorship

CrimsonShrike

## Changelog

:cl: CrimsonShrike
rscadd: Swimming.
/:cl:

### CODE REVIEW

File: code/modules/multiz/movement.dm [hunk 1]
```diff
@@ -50,6 +50,11 @@
 	if(species && species.can_overcome_gravity(src))
 		return 1
 	else
+		var/turf/T = loc
+		if(((T.get_physical_height() + T.get_fluid_depth()) >= FLUID_DEEP) || T.get_fluid_depth() >= FLUID_MAX_DEPTH)
```
Comment (MistakeNot4892) [inline code/modules/multiz/movement.dm:54]: Fluid depth should be independent of the physical height of the turf - that value is used for flooding calc only.

File: code/modules/multiz/movement.dm [hunk 2]
```diff
@@ -173,9 +185,19 @@
 		return species.can_fall(src)
 
 /atom/movable/proc/handle_fall(var/turf/landing)
+	var/turf/previous = get_turf(loc)
 	forceMove(landing)
 	if(locate(/obj/structure/stairs) in landing)
 		return 1
+	else if(landing.get_fluid_depth() >= FLUID_DEEP)
```
Comment (MistakeNot4892) [inline code/modules/multiz/movement.dm]: Unneeded `else`.

File: code/modules/multiz/movement.dm [hunk 3]
```diff
@@ -143,6 +148,13 @@
 		for(var/atom/A in below)
 			if(!A.CanPass(src, location_override))
 				return FALSE
+		
+		//We cannot sink if we can swim
+		if(location_override.get_fluid_depth() >= FLUID_DEEP)
+			if(below == loc) //We are checking above, 
+				if(!(below.get_fluid_depth() >= 0.95 * FLUID_MAX_DEPTH)) //No salmon skipping up a stream of falling water
```
Comment (MistakeNot4892) [inline code/modules/multiz/movement.dm]: Could collapse the second `if` into the first.

**Comment by MistakeNot4892** [comment]
Generally looks fine, just needs some small tweaks.

File: code/modules/multiz/movement.dm (hunk 1 above)
Comment (CrimsonShrike) [reply code/modules/multiz/movement.dm:54]: This is so if you have a deep pool you cannot somehow swim up towards the ceiling

**Comment by PlayerDeer** [comment]
Now someone sprite high detail swimming animations

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**